src/main/java/de/tum/in/www1/artemis/web/rest/ApollonDiagramResource.java

```diff
@@ -98,8 +98,8 @@ public ApollonDiagramResource(ApollonDiagramRepository apollonDiagramRepository)
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     public ResponseEntity<ApollonDiagram> getApollonDiagram(@PathVariable Long id) {
         log.debug("REST request to get ApollonDiagram : {}", id);
-        ApollonDiagram apollonDiagram = apollonDiagramRepository.findById(id).get();
-        return ResponseUtil.wrapOrNotFound(Optional.ofNullable(apollonDiagram));
```

Comment:
This would have returned a 500 error due to the Optional.get() instead of a 404 which was described in the javadoc